src/public/js/util.js

```diff
@@ -38,6 +38,12 @@ var UTIL = (function() {
      0xb4, 0x7c, 0xb8, 0x26, 0x77, 0x99, 0xe3, 0xa5,
      0x67, 0x4a, 0xed, 0xde, 0xc5, 0x31, 0xfe, 0x18,
      0x0d, 0x63, 0x8c, 0x80, 0xc0, 0xf7, 0x70, 0x07]);
+  log3.get = function(x) {
```

Comment:
Ah awesome! From what I had been reading on JS I suspected you would get undefined in such a situation :) Glad to see my understanding is not off base! Thanks for the pointers.

This discussion definitely helped me understand the object system of JS more. I am surprised as I don't think I have ever worked in a language for which you could add new fields to an object after declaring it. Seems to me that all objects are basically implemented as a map with the keys being the names of its fields and the values being whatever is stored in the field. For arrays I suspect underneath there is a data field and doing arr[0] is like saying arr['data'][0] really.

Not sure if you know too much about this but if so let me know if I am on or off base here.